### ISSUE TITLE
why my podstat.ContainerStats.Logs.UsedBytes continued rise?

### ISSUE THREAD

**Issue by kunkunhaohao** [kubernetes/kubernetes#119691]
### What happened?

In k8s v1.19.4, I tried to configure the ephemeral-storage field in the pod yaml file, and after a while I found that my pod was evicted.
I analyzed the code to determine what was wrong with my container, and then I found that ephemeral-storage included pod log files as part of ephemeral-storage during calculation, but it seems that k8s had an error in calculating logs. It doubles the log usage every minute and then evicts the pod until the result is greater than ephemeral-storage.
log file path is: /var/log/pods/...

I try to add logs in pkg/kubelet/stats/helper.go calcEphemeralStorage to see if logs.UsedBytes keeps increasing.
func calcEphemeralStorage(containers []statsapi.ContainerStats, volumes []statsapi.VolumeStats, rootFsInfo *cadvisorapiv2.FsInfo,
	* * *
	for _, container := range containers {
		klog.Infof("print 2 contanier name is: %v, log is: %v", container.Name, *diskUsage(container.Logs))
		addContainerUsage(result, &container, isCRIStatsProvider)
		klog.Infof("print 3 contanier name is: %v, log is: %v", container.Name, *diskUsage(container.Logs))
	}
* * *
}
I0731 20:48:08.760675   54318 helper.go:364] print 2 contanier name is: sidecar-iptable, log is: {{167936 0} {<nil>}  BinarySI}
I0731 20:48:08.760689   54318 helper.go:366] print 3 contanier name is: sidecar-iptable, log is: {{167936 0} {<nil>}  BinarySI}
I0731 20:48:08.760697   54318 helper.go:364] print 2 contanier name is: notebook, log is: {{569344 0} {<nil>}  BinarySI}
I0731 20:48:08.760708   54318 helper.go:366] print 3 contanier name is: notebook, log is: {{569344 0} {<nil>}  BinarySI}

I0731 20:48:18.780797   54318 helper.go:364] print 2 contanier name is: notebook, log is: {{569344 0} {<nil>}  BinarySI}
I0731 20:48:18.780807   54318 helper.go:366] print 3 contanier name is: notebook, log is: {{569344 0} {<nil>}  BinarySI}
I0731 20:48:18.780818   54318 helper.go:364] print 2 contanier name is: sidecar-iptable, log is: {{167936 0} {<nil>}  BinarySI}
I0731 20:48:18.780829   54318 helper.go:366] print 3 contanier name is: sidecar-iptable, log is: {{167936 0} {<nil>}  BinarySI}

I0731 20:48:28.800686   54318 helper.go:364] print 2 contanier name is: notebook, log is: {{569344 0} {<nil>}  BinarySI}
I0731 20:48:28.800693   54318 helper.go:366] print 3 contanier name is: notebook, log is: {{569344 0} {<nil>}  BinarySI}
I0731 20:48:28.800698   54318 helper.go:364] print 2 contanier name is: sidecar-iptable, log is: {{167936 0} {<nil>}  BinarySI}
I0731 20:48:28.800703   54318 helper.go:366] print 3 contanier name is: sidecar-iptable, log is: {{167936 0} {<nil>}  BinarySI}

I0731 20:48:38.819571   54318 helper.go:364] print 2 contanier name is: notebook, log is: {{569344 0} {<nil>}  BinarySI}
I0731 20:48:38.819640   54318 helper.go:366] print 3 contanier name is: notebook, log is: {{569344 0} {<nil>}  BinarySI}
I0731 20:48:38.819652   54318 helper.go:364] print 2 contanier name is: sidecar-iptable, log is: {{167936 0} {<nil>}  BinarySI}
I0731 20:48:38.819660   54318 helper.go:366] print 3 contanier name is: sidecar-iptable, log is: {{167936 0} {<nil>}  BinarySI}

I0731 20:48:48.837492   54318 helper.go:364] print 2 contanier name is: sidecar-iptable, log is: {{167936 0} {<nil>}  BinarySI}
I0731 20:48:48.837498   54318 helper.go:366] print 3 contanier name is: sidecar-iptable, log is: {{167936 0} {<nil>}  BinarySI}
I0731 20:48:48.837503   54318 helper.go:364] print 2 contanier name is: notebook, log is: {{569344 0} {<nil>}  BinarySI}
I0731 20:48:48.837509   54318 helper.go:366] print 3 contanier name is: notebook, log is: {{569344 0} {<nil>}  BinarySI}

I0731 20:48:58.856753   54318 helper.go:364] print 2 contanier name is: sidecar-iptable, log is: {{335872 0} {<nil>}  BinarySI}
I0731 20:48:58.856759   54318 helper.go:366] print 3 contanier name is: sidecar-iptable, log is: {{335872 0} {<nil>}  BinarySI}
I0731 20:48:58.856764   54318 helper.go:364] print 2 contanier name is: notebook, log is: {{1138688 0} {<nil>}  BinarySI}
I0731 20:48:58.856773   54318 helper.go:366] print 3 contanier name is: notebook, log is: {{1138688 0} {<nil>}  BinarySI}

### What did you expect to happen?

配置了ephemeral-storage的pod不应该被驱逐

### How can we reproduce it (as minimally and precisely as possible)?

Do not configure ephemeral-storage for pods.

### Anything else we need to know?

I compared k8s v1.19.4 with k8s v1.25.3, and I found that 1.25.3 did not have this problem.
In addition, I found that v1.25.3 refactored some of the code https://github.com/kubernetes/kubernetes/commit/acb43c7c4ac565eb2b51f98dcce32b5879383a19

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```
v1.19.4
</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by aojea** [comment]
1.19 is end of life  , so it should not receive patches, you have to update to a supported version

/kind support 
/sig node

**Comment by kunkunhaohao** [comment]
> I just want to know why this is happening, can you explain it from the code level?

**Comment by mmiranda96** [comment]
Not sure if there's any further investigation we can do, as it was fixed in 1.25.

/close
